.github/workflows/schema-check.yml

```diff
@@ -0,0 +1,26 @@
+name: DCR Schema Check
+on:
+  push:
+    paths-ignore:
+      - "apps-rendering/**"
+
+jobs:
+  build_check:
+    name: DCR Schema Check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Install Node
+        uses: guardian/actions-setup-node@main
+
+      # Cache npm dependencies using https://github.com/bahmutov/npm-install
+      - uses: bahmutov/npm-install@v1
+
+      - name: Install
+        run: yarn
+
+      - name: Run check-schema script
```

Comment:
Maybe bikeshedding, but for a more streamlined DevX, what about committing as a bot rather than requiring a human to do the task? Something similar to [this](https://lannonbr.com/blog/2019-12-09-git-commit-in-actions).

So a workflow that follows:
* Check schema
* On pass: no-op
* On fail: gen-schema & commit
* Handle any gen-schema errors

Reply:
I'm actually really hesitant to do this - 

The schemas generated are verified when articles are loaded, for example - if a new required property was added to index.d.ts, and it turned out that it wasn't always provided from Frontend, DCR would reject the article, 500ing, as it doesn't the schema. 

Therefor I think making sure that people are being purposeful when updating the schema, and are aware of the impact it can have is really important.

Theoretically our tests will catch these cases, but it would feel weird to have a bot commit changes that would make tests fail.

Reply:
Thinking about the workflow of this, wouldn't it start with frontend? e.g.
* update frontend types
* update types in DCR
* gen-schema

I'd say `gen-schema` is just a step we have to do once the thinking has been done, as we should be thinking hard if things are going to break when updating `frontend`.

This also exposes the tight coupling we have between the two apps, and that we're more at risk of breaking this in frontend e.g. removing a required field from the models there. I wonder if there something thinking to be done in exposing FE interfaces from there, especially as it becomes more of an API service.

Reply:
For day-to-day dev it starts with Frontend, but as we do migration things with fronts - there are times we start typing things in DCR which already exist in Fronts, and missing schema updates can lead to [mistakes](https://github.com/guardian/dotcom-rendering/pull/4884).

This certainly isn't an end-game in DevX or the relationship between DCR and Frontend, but it will ensure that we always have an up-to-date schema based on our types in DCR :)

Reply:
(It would be awesome to take a closer look at the type definitions between frontend and DCR and improve the workflows there!)

Reply:
> (It would be awesome to take a closer look at the type definitions between frontend and DCR and improve the workflows there!)

This is definitely what I was getting at...